tools/generate_intrinsics.py

```diff
@@ -0,0 +1,119 @@
+import os
+import re
+import sys
+import subprocess
+from os import walk
+
+
+LLVM_PATH = llvm_path = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)),
+    "llvm-project",
+)
+
+def run_command(command, cwd=None):
+    p = subprocess.Popen(command, cwd=cwd)
+    if p.wait() != 0:
+        print("command `{}` failed...".format(" ".join(command)))
+        sys.exit(1)
+
+
+def clone_llvm_repository():
+    if os.path.exists(LLVM_PATH):
+        while True:
+            choice = input("There is already a llvm-project folder, do you want to update it? [y/N]")
+            if choice == "" or choice.lower() == "n":
+                print("Skipping repository update.")
+                return
+            elif choice.lower() == "y":
+                print("Updating repository...")
+                run_command(["git", "pull", "origin"], cwd="llvm-project")
+                return
+            else:
+                print("Didn't understand answer...")
+    print("Cloning LLVM repository...")
+    run_command(["git", "clone", "https://github.com/llvm/llvm-project", "--depth", "1", LLVM_PATH])
```

Comment:
Maybe use a blobless clone with a sparse checkout? Something like

```sh
$ git clone https://github.com/llvm/llvm-project.git --filter=tree:0 --no-checkout
$ cd llvm-project
$ git sparse-checkout init
$ git sparse-checkout set add llvm/include/llvm/IR
```

This downloads 125MB resulting in a 160MB checkout without github spending any time to create a new packfile from scratch as `--depth 1` requires. For reference `--depth 1` downloads 170MB resulting in a 1.6GB checkout while taking longer.

Reply:
I didn't know about this. Let me give it a try.

Reply:
The end result is great! Thanks for this. :)